extensions/big-data-cluster/src/bigDataCluster/utils.ts

```diff
@@ -210,3 +211,46 @@ export function getHealthStatusDot(healthStatus?: string): string {
 			return '•';
 	}
 }
+
+
+interface RawEndpoint {
+	serviceName: string;
+	description?: string;
+	endpoint?: string;
+	protocol?: string;
+	ipAddress?: string;
+	port?: number;
+}
+
+export interface IEndpoint {
+	serviceName: string;
+	description: string;
+	endpoint: string;
+	protocol: string;
+}
+
+export function getClusterEndpoints(serverInfo: azdata.ServerInfo): IEndpoint[] {
+	let endpoints: RawEndpoint[] = serverInfo.options[constants.clusterEndpointsProperty];
+	if (!endpoints || endpoints.length === 0) { return []; }
+
+	return endpoints.map(e => {
+		// If endpoint is missing, we're on CTP bits. All endpoints from the CTP serverInfo should be treated as HTTPS
+		let endpoint = e.endpoint ? e.endpoint : `https://${e.ipAddress}:${e.port}`;
+		let updatedEndpoint: IEndpoint = {
+			serviceName: e.serviceName,
+			description: e.description,
+			endpoint: endpoint,
+			protocol: e.protocol
+		};
+		return updatedEndpoint;
+	});
+}
+
+export function getControllerEndpoint(serverInfo: azdata.ServerInfo): string | undefined {
+	let endpoints = getClusterEndpoints(serverInfo);
+	if (endpoints) {
+		let index = endpoints.findIndex(ep => ep.serviceName.toLowerCase() === constants.controllerEndpointName.toLowerCase());
```

Comment:
return undefined if this is false, right now you're not returning anything in that case.

Reply:
I'm returning undefined in the next line down? This function returns -1 if index isn't found, and then I return undefined as suggested. I'm going to close this as bydesign?